### ISSUE TITLE
[ci] use nix caching for actions cache

### ISSUE THREAD

**Issue by VTimofeenko** [xremap/nix-flake#52]

**Comment by musjj** [comment]
Would love to have this too.

There's instructions here: https://nix.dev/tutorials/nixos/continuous-integration-github-actions. Shouldn't be hard to set it up.

Also, it's free for open-source projects (up to 5GB).

**Comment by VTimofeenko** [comment]
I was actually thinking about magic-nix-cache. 

Currently the build action is configured to run <=> the package definition changes, and most of the development in this repo is done on modules, so there is little benefit to be had in this for now.

It would probably be more relevant when qemu-based tests are done in #16

**Comment by musjj** [comment]
I don't think end-users can use the magic-nix-cache as a binary cache. It's only meant to be used to speed up the CI.
I thought the issue was about providing a binary cache so that users don't have to build the program locally.

**Comment by VTimofeenko** [comment]
Yeah, I don't think github cache can be made public. End-user facing cache would be #29

**Comment by musjj** [comment]
Oh thanks, didn't notice that issue.
Btw, does this concern:
> cachix requires specifying secrets on the repo level and ownership of one more system. I'd rather keep this project self-contained on github.

also applies to the end-user facing cache? Because I don't think it'd be possible to provide a nix binary cache that is self-contained on GitHub.
